vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
+          }
         }
+        return false;
+      })
 
-      } else {
-        grid.$server.setParentRequestedRange(firstIndex, size, parentKey);
-      }
-    }
-
-    grid.$connector.fetchPage = function(fetch, page, parentKey) {
-      // Determine what to fetch based on scroll position and not only
-      // what grid asked for
-
-      // The buffer size could be multiplied by some constant defined by the user,
-      // if he needs to reduce the number of items sent to the Grid to improve performance
-      // or to increase it to make Grid smoother when scrolling
-      let start = grid._virtualStart;
-      let end = grid._virtualEnd;
-      let buffer = end - start;
-
-      let firstNeededIndex = Math.max(0, start + grid._vidxOffset - buffer);
-      let lastNeededIndex = Math.min(end + grid._vidxOffset + buffer, grid._effectiveSize);
-
-      let firstNeededPage = page;
-      let lastNeededPage = page;
-      for(let idx = firstNeededIndex; idx <= lastNeededIndex; idx++) {
-        firstNeededPage = Math.min(firstNeededPage, grid.$connector._getPageIfSameLevel(parentKey, idx, firstNeededPage));
-        lastNeededPage = Math.max(lastNeededPage, grid.$connector._getPageIfSameLevel(parentKey, idx, lastNeededPage));
-      }
+      grid.$connector.flushParentRequests = tryCatchWrapper(function() {
+        let pendingFetches = parentRequestQueue.splice(0, parentRequestBatchMaxSize);
 
-      let firstPage = Math.max(0,  firstNeededPage);
-      let lastPage = (parentKey !== root) ? lastNeededPage: Math.min(lastNeededPage, Math.floor(grid.size / grid.pageSize));
-      let lastRequestedRange = lastRequestedRanges[parentKey];
-      if(!lastRequestedRange) {
-        lastRequestedRange = [-1, -1];
-      }
-      if (lastRequestedRange[0] != firstPage || lastRequestedRange[1] != lastPage) {
-        lastRequestedRange = [firstPage, lastPage];
-        lastRequestedRanges[parentKey] = lastRequestedRange;
-        let count = lastPage - firstPage + 1;
-        fetch(firstPage * grid.pageSize, count * grid.pageSize);
-      }
-    }
+        if(pendingFetches.length) {
+            grid.$server.setParentRequestedRanges(pendingFetches);
+            return true;
+        }
+        return false;
+      })
+
+      grid.$connector.beforeParentRequest = tryCatchWrapper(function(firstIndex, size, parentKey) {
+        if(parentRequestDelay > 0) {
+          // add request in queue
+          parentRequestQueue.push({
+            firstIndex: firstIndex,
+            size: size,
+            parentKey: parentKey
+          });
+
+          if(!parentRequestDebouncer) {
+              grid.$connector.flushQueue(
+                (debouncer) => parentRequestDebouncer = debouncer,
+                () => grid.$connector.hasParentRequestQueue(),
+                () => grid.$connector.flushParentRequests(),
+                (action) => Debouncer.debounce(parentRequestDebouncer, timeOut.after(parentRequestDelay), action)
+                );
```

Comment:
Fix indentation. Note also that line 332 should be on the same level as first line

Reply:
Updated.